momo_modules/multi_sensor_module/src/app_main.c

```diff
@@ -2,52 +2,75 @@
 
 #include "platform.h"
 #include "watchdog.h"
-#include "sensor_defines.h"
 #include "port.h"
 #include "sample.h"
 #include "mib12_api.h"
 #include "log.h"
 #include "digital_amp.h"
 #include "watchdog.h"
+#include "ioc.h"
 #include "state.h"
 #include "pulse.h" 
 #include "alarm_repeat_times.h"
 
 MultiSensorState state;
+
 extern unsigned int adc_result;
+extern uint16_t counter;
+extern uint16_t periods;
 
 void task(void)
 {
-	while (state.acquire_pulse)
-	{
+	wdt_disable();
+
+	//If we slept for an increment, sample the number of pulses
+	if (nTO==0)
 		pulse_sample();
-		state.acquire_pulse = 0;
 
-		if ( state.push_pending )
-		{
-			state.push_pending = 0;
+	if (state.push_pending)
+	{
+		uint16_t correction_factor = 1;
+		state.push_pending = 0;
+
+		/*
+		 * We report data every minute, and we sample for 0.1 seconds
+		 * per time.  So we need to correct our sampled counter for the
+		 * ratio between the period of time that we are covering (60 seconds)
+		 * and the amount of time we were sampling for (periods*0.1 seconds)
+		 */
+
+		if (periods > 0)
```

Comment:
When do we expect this to happen?  Won't it throw off the correction_factor calculation significantly if periods is ever 0 at this point?